src/eth_provider/provider.rs

```diff
@@ -553,7 +553,7 @@ where
         let pending_transaction = self.database.get_one::<StoredPendingTransaction>(filter.clone(), None).await?;
 
         // Number of retries for the transaction (0 if it's a new transaction)
-        let retries = pending_transaction.as_ref().map(|tx| tx.retries).unwrap_or_default();
+        let retries = pending_transaction.as_ref().map(|tx| tx.retries + 1).unwrap_or_default();
```

Comment:
Aaaaah yes good catch, in the lines after we have two

`pending_transaction.retries + 1`

Can't they be replaced by 

`retries`

?